### ISSUE TITLE
[Feature] Support validating PWA installation and offline functionality

### ISSUE THREAD

**Issue by TanayParikh** [microsoft/playwright#19677]
I'd like the following to work:

```csharp
        var page = await browser.NewPageAsync();
        await page.GotoAsync(URI_TO_PWA_APP, new() { WaitUntil = WaitUntilState.NetworkIdle });
        // have some playwright command here to "install" the PWA
        await page.GotoAsync("about:blank");
        await browser.SetOfflineAsync(true);
        await page.GotoAsync(URI_TO_PWA_APP); // currently fails here as we're "offline" and just navigating to a page (as we do on line 2) doesn't install the PWA
```

**Comment by pavelfeldman** [comment]
When you are saying 'install PWA' you probably mean "allow enough time for service worker to install and to populate its offline cache". We don't have a doc for this in .NET, you can look at https://playwright.dev/docs/next/service-workers-experimental#accessing-service-workers-and-waiting-for-activation to get an idea. Once service worker is in place, you can evaluate code in it to check if its caches are populated (using waitForFunction or alike). After that, loading will be served from the service worker and can be performed offline.

**Comment by pavelfeldman** [comment]
Closing as per above, please feel free to open a new issue if this does not cover your use case.
